app/models/event.rb

```diff
@@ -82,9 +82,14 @@ def self.format_date(type)
   end
 
   #scope
-  scope :recent_events, -> {order(created_at: :DESC).limit(12)}
-  scope :featured_events, -> {order(created_at: :DESC).limit(2)}
-  scope :popular_events, -> {order(created_at: :DESC).limit(3)}
+  scope :recent_events, -> { order(created_at: :DESC).limit(12) }
+  scope :featured_events, -> { order(created_at: :DESC).limit(12) }
+  scope :popular_events, -> {
+    find_by_sql("Select events.*, COUNT(bookings.event_id) AS num from events INNER JOIN bookings WHERE bookings.event_id = events.id GROUP BY events.id ORDER BY num")
+  }
+  scope :upcoming_events, -> { where("start_date >= ?", Time.zone.now).
+                              order("start_date ASC").limit(12)
+                            }
```

Comment:
`end` at 92, 28 is not aligned with `-> { where("start_date >= ?", Time.zone.now).` at 90, 26 or `scope :upcoming_events, -> { where("start_date >= ?", Time.zone.now).` at 90, 2.